utils.go

```diff
@@ -336,6 +345,12 @@ func CborReadHeaderBuf(br io.Reader, scratch []byte) (byte, uint64, error) {
 		return 0, 0, err
 	}
 
+	defer func() {
+		if err == io.EOF {
+			err = io.ErrUnexpectedEOF
+		}
+	}()
```

Comment:
This has no effect unless we use a named return value for the error.

Reply:
Hm... I'm concerned that the above error will actually _shadow_ the return error. I think this is a more general bug.

Reply:
We need to fix https://github.com/whyrusleeping/cbor-gen/pull/68 then apply the same changes here.

Reply:
Left a comment on #68

Reply:
Yep, you're correct.